### PR TITLE
 Fix issue with serialization when using a Table's subclass 

### LINKED ISSUE #7181: Table's subclass break with serialization
With the serialization of astropy objects to FITS (#6912), I just noticed that it breaks the use of our `Table` subclass. When writing and reading back a FITS file with this subclass, `io.registry.read` now fails because the reader return a `QTable`, for which my class is not a subclass.
I don't see a way to disable the serialization, or to not use it while reading. Any idea @taldcroft ?

### PR DESCRIPTION
**Pull request by saimn**
Fixes #7181, as suggested by @taldcroft

### CODE REVIEW

**Comment by saimn** [comment]
ping @taldcroft

File: astropy/io/tests/test_registry.py [hunk 1]
```diff
@@ -410,3 +402,23 @@ class MyTable(Table):
...
+        mt = MTable([[1, 2.5]], names=['a'])
+        mt['a'].unit = u.m
+        mt['a'].format = '.4f'
+        mt['a'].description = 'hello'
+
+        testfile = str(tmpdir.join('junk.fits'))
+        mt.write(testfile, overwrite=True)
+
+        t = MTable.read(testfile)
+        assert np.all(mt == t)
```
Comment (taldcroft) [inline astropy/io/tests/test_registry.py]: You should check that the `mt['a']` attributes are the same.

Turns out that `mt == t` does not check column attributes, so apart from the fact that the read operation does not fail outright, you don't know that the unit survived serialization.

File: CHANGES.rst [hunk 2]
```diff
@@ -291,6 +291,8 @@ astropy.io.misc
 astropy.io.registry
 ^^^^^^^^^^^^^^^^^^^
 
+- Fix reading files with serialized metadata when using a Table's subclass. [#7213]
```
Comment (taldcroft) [inline CHANGES.rst]: Should just be `... using a Table subclass`.

**Comment by saimn** [comment]
@taldcroft - Thanks for the comments, it should be ok now.

**Comment by saimn** [comment]
Ah, it fails for several builds, I guess for the ones that do not have pyyaml.

**Comment by taldcroft** [comment]
Thanks @saimn !

**Pull request merged**